### DISCUSSION TITLE
Allow requiresConfirmation() for default modal footer action buttons

### DISCUSSION THREAD

**Discussion by lpeterke** [filamentphp/filament#13366] [Ideas]
Regarding this part of the docs: https://filamentphp.com/docs/3.x/actions/modals#modifying-a-default-modal-footer-action-button

It would be great to be able to use a confirmation modal on top of an already opened modal.

**Use Case:**
For my OrderResource I am using a modal with a form instead of a create page, so I can place this modal on multiple places around my app. The form of the modal is quite large and uses a wizard with multiple steps. Because of this I would like to ask for confirmation, if the user really wants to close the create order modal, as this would clear the form.

**Comment by micraux** [comment]
Is [this](https://filamentphp.com/docs/3.x/actions/modals#opening-another-modal-from-an-extra-footer-action) what you are looking for?

**Reply by lpeterke** [reply]
I dont think so. The extra modal footer action would be a different nested action.

If I would disable the modal submit action I could use the extra modal footer action to achieve the UX that I want but I am not sure that I am able to cycle back and trigger the submit of the parent action after I proceeded the confirmation modal, as the filament docs only mention a cancelParentAction() method 🤔

**Reply by wijaksanapanji** [reply]
Did you find solution for this?
I also need to add confirmation modal on top of the form modal

**Reply by mahjoubkzbr** [reply]
Anyone found a solution yet?

**Reply by Syzgyn** [reply]
@kouzkoko I just managed to figure this out using `cancelParentActions`
```
Action::make('modalForm')
            ->form([...])
            ->modalSubmitAction(false)
            ->extraModalFooterActions(fn (Action $action) => [
                Action::make('submit')
                    ->requiresConfirmation()
                    ->cancelParentActions('modalForm')
                    ->action(...)
            ]);
```
It's a little hacky, but inside `action()` you can call your submit function or do whatever else you need to.
